### PR TITLE
About us

### PR DESCRIPTION
**Pull request by jorgebaralt**
# Description

- Description of the changes done in the pulled request for this branch.

## Type of the change (mark any which applies)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: src/components/UI/FounderCard/index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { withTheme } from 'styled-components';
+
+const FounderCard = (props) => {
+  const { founder } = props;
+  return (
+    <CardOutside>
+      <CircularImage src="https://www.w3schools.com/howto/img_avatar.png" alt="Avatar" />
```
Comment (rmolinamir) [inline src/components/UI/FounderCard/index.js:12]: Should be:

```
{/* TODO: Remove Placeholder */}
<CircularImage src="https://www.w3schools.com/howto/img_avatar.png" alt="Avatar" />
```

File: .eslintrc [hunk 2]
```diff
@@ -27,175 +21,62 @@
...
-    "global-require": [
-      0
-    ],
-    "import/no-extraneous-dependencies": [
-      0
-    ],
-    "comma-spacing": [
-      1
-    ],
-    "arrow-parens": [
```
Comment (rmolinamir) [inline .eslintrc:142]: Let's keep this one

File: src/components/UI/LazyImport/index.js [hunk 3]
```diff
@@ -5,7 +5,7 @@ import PropTypes from 'prop-types';
 /**
  * After the lazy imported component loads, delay the promise to increase smoothness.
  */
-const LazyImport = props => {
+const LazyImport = (props) => {
```
Comment (rmolinamir) [inline src/components/UI/LazyImport/index.js:8]: This is an eslint rule:

`Unexpected parentheses around single function argument.eslint(arrow-parens)`

File: src/containers/AboutUs/index.js [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+    if (founderContainerRef && founderContainerRef.current) {
+      founderContainerRef.current.scrollIntoView({
+        behavior: 'smooth',
+        block: 'end',
+        inline: 'nearest'
+      });
+    }
+  };
+
+  const foundersData = [
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: We can use `useMemo` for this array for optimization.

File: src/containers/AboutUs/index.js [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+import FounderCard from '../../components/UI/FounderCard';
+import JoinUsImage from '../../static/images/backgrounds/join-us.jpg';
+import HookedParallax from '../../components/UI/HookedParallax';
+
+const AboutUs = (props) => {
+  // Hooks
+  const founderContainerRef = useRef(null);
+
+  // Functions
+  const ArrowDownClickHandler = () => {
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: `useCallback` to avoid re-renders.

File: src/containers/AboutUs/index.js [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+      color: props.theme.brandOrange
+    }
+  ];
+
+  return (
+    <Container>
+      <HeaderStyle>
+        <HeaderContainer>
+          <AppText color={props.theme.whiteColor} fontSize="42" fontWeight="bold">
+            Devoted to provide{' '}
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js:65]: Maybe more readable to use a template literal:

```
{`Devoted to provide ${<SpanText {...}>{...}</SpanText>}...`}
```

Or just:


```
{`Devoted to provide `}
```

File: src/containers/AboutUs/index.js [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+// Components
+const Container = styled.div`
+  color: ${props => props.theme.lightDarkColor};
+  transition: all ease 200ms;
+`;
+
+const HeaderContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: 74vh;
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: Why 74? Lol

File: src/containers/AboutUs/index.js [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+  height: ${props => `${props.size}px`};
+  width: ${props => `${props.size}px`};
+  border-radius: 30px;
+  transform: ${props => `rotateY(0deg) rotate(${props.rotate}deg)`};
+  background-color: ${props => props.color};
+  opacity: 0.7;
+`;
+
+const FounderContainer = styled.div`
+  height: 100vh;
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: There's a CSS bug when using `100vh` on mobile, it's better to use `window.innerHeight`, check out the homepage, specifically the `Container` component to see how I'm handling this over there.

File: src/containers/App/index.js [hunk 9]
```diff
@@ -20,6 +20,11 @@ const RESOLVED_CALLBACK_DELAY = process.env.NODE_ENV === 'development' ? LOADER_
 const App = () => {
   const [isLoading, setIsLoading] = React.useState(true);
 
+  React.useEffect(() => {
```
Comment (rmolinamir) [inline src/containers/App/index.js:23]: What is this for?

File: src/containers/Homepage/index.js [hunk 10]
```diff
@@ -27,7 +27,7 @@ import {
   StyledHeroValueProposition
 } from './components';
 
-const Homepage = props => {
+const Homepage = (props) => {
```
Comment (rmolinamir) [inline src/containers/Homepage/index.js:30]: eslint rule mentioned above!

**Review by rmolinamir — CHANGES_REQUESTED** [review]
Small changes 👍

File: .eslintrc (hunk 2 above)
Comment (jorgebaralt) [reply .eslintrc:142]: is not removed just formatted

File: src/components/UI/LazyImport/index.js (hunk 3 above)
Comment (jorgebaralt) [reply src/components/UI/LazyImport/index.js:8]: always presenting parens make refactoring easier: when you need to add another arg you don't need to add parens (cause they are present already).

File: src/containers/AboutUs/index.js (hunk 4 above)
Comment (jorgebaralt) [reply src/containers/AboutUs/index.js]: fixed

File: src/containers/AboutUs/index.js (hunk 5 above)
Comment (jorgebaralt) [reply src/containers/AboutUs/index.js]: added

File: src/containers/App/index.js (hunk 9 above)
Comment (jorgebaralt) [reply src/containers/App/index.js:23]: Some components make the app X scrollable. this locks the screen making sure it is not allowed

File: src/containers/Homepage/index.js (hunk 10 above)
Comment (jorgebaralt) [reply src/containers/Homepage/index.js:30]: always presenting parens make refactoring easier: when you need to add another arg you don't need to add parens (cause they are present already).

File: src/components/UI/FounderCard/index.js [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+  align-items: center;
+  border: 1px solid ${props => props.theme.darkColor};
+  background-color: ${props => props.theme.darkColor};
+  border-radius: 10px;
+  overflow: hidden;
+  padding: 100px 20px 20px 20px;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
+`;
+
+const AppText = styled.p`
```
Comment (rmolinamir) [inline src/components/UI/FounderCard/index.js]: Should be in a folder under `components` called `fontSystem`. And then this should be a component called `P`:

```
const P = styled.p`
  color: ${props => props.color || '#000'};
  font-size: 15px;
  margin-bottom: ${props => (props.nomargin ? 0 : '10px')};
  font-style: ${({ italic }) => (italic ? 'italic' : 'normal')};
`
```

We can then do a bunch of components, such as:

```
const H1 = styled.h1`
  color: ${props => props.color || props.theme.primary};
  font-size: 20px;
  line-height: 30px;
`;

const H2 = styled.h2`
  color: ${props => props.color || props.theme.primaryColor};
  font-size: 16px;
  line-height: 26px;
`;
```

File: src/components/UI/HookedParallax/index.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import React, { useRef, useState, useLayoutEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import { useTranslateContent } from '../../../utils/hooks/useTranslateContent';
+
+// TODO: check mobile
```
Comment (rmolinamir) [inline src/components/UI/HookedParallax/index.js:6]: We do this easily in Next.js.

In the browser, we'll need to check user agent in a top-level function.

Use this:

```
export const isMobile = () => Boolean(/Mobi|Android/i.test(navigator.userAgent))
}
```

File: src/containers/AboutUs/index.js [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+      window.scroll({
+        top: 0,
+        left: 0,
+        behavior: 'smooth'
+      });
+    } catch (error) {
+      // just a fallback for older browsers
+      window.scrollTo(0, 0);
+    }
+  });
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: Missing dependency list! If on during mount, then add an empty array.

File: src/containers/AboutUs/index.js [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+import HookedParallax from '../../components/UI/HookedParallax';
+
+const { innerHeight } = window;
+
+const AboutUs = (props) => {
+  // Hooks
+  const founderContainerRef = useRef(null);
+
+  // On mount
+  useEffect(() => {
```
Comment (rmolinamir) [inline src/containers/AboutUs/index.js]: Make this a custom hook!

File: src/containers/App/index.js [hunk 15]
```diff
@@ -20,6 +20,11 @@ const RESOLVED_CALLBACK_DELAY = process.env.NODE_ENV === 'development' ? LOADER_
...
+    document.body.style.overflowX = 'hidden';
+    return () => { document.body.style.overflowX = 'visible'; };
+  });
```
Comment (rmolinamir) [inline src/containers/App/index.js]: Missing dependency list

File: src/utils/hooks/useTranslateContent/index.js [hunk 16]
```diff
@@ -0,0 +1,35 @@
+import { useState, useEffect, useCallback } from 'react';
+import _ from 'lodash';
+
+/* 
+	`useTranslateContent` takes a multiplier and returns the translateY value
+*/
+
+export const useTranslateContent = (multipliersY, { startingY, componentHeight }) => {
```
Comment (rmolinamir) [inline src/utils/hooks/useTranslateContent/index.js:8]: Should we rename to `useElementTranslate`?

**Review by rmolinamir — CHANGES_REQUESTED** [review]

File: src/components/UI/AppText/index.js [hunk 17]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Typography from '@material-ui/core/Typography';
```
Comment (rmolinamir) [inline src/components/UI/AppText/index.js]: `Typography` is unused tho.

I was thinking transcribing the Typography from:

```
      <Typography variant="h1" component="h2" gutterBottom>
        h1. Heading
      </Typography>
      <Typography variant="h2" gutterBottom>
        h2. Heading
      </Typography>
      <Typography variant="h3" gutterBottom>
        h3. Heading
      </Typography>
      <Typography variant="h4" gutterBottom>
        h4. Heading
      </Typography>
      <Typography variant="h5" gutterBottom>
        h5. Heading
      </Typography>
      <Typography variant="h6" gutterBottom>
        h6. Heading
      </Typography>
      <Typography variant="subtitle1" gutterBottom>
        subtitle1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur
      </Typography>
      <Typography variant="subtitle2" gutterBottom>
        subtitle2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur
      </Typography>
      <Typography variant="body1" gutterBottom>
        body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur
        unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam
        dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
      </Typography>
      <Typography variant="body2" gutterBottom>
        body2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos blanditiis tenetur
        unde suscipit, quam beatae rerum inventore consectetur, neque doloribus, cupiditate numquam
        dignissimos laborum fugiat deleniti? Eum quasi quidem quibusdam.
      </Typography>
      <Typography variant="button" display="block" gutterBottom>
        button text
      </Typography>
      <Typography variant="caption" display="block" gutterBottom>
        caption text
      </Typography>
      <Typography variant="overline" display="block" gutterBottom>
        overline text
      </Typography>
```

To components like this:

```
      ...
      <Caption display="block" gutterBottom>
        caption text
      </Caption >
      <Overline display="block" gutterBottom>
        overline text
      </Overline >
      ...
```

And then Caption would look like:

```
const Caption = props => (
      <Typography variant="caption" {...props}>
        caption text
      </Typography>
)
const StyledCaption = styled(Caption)`
    font-size: XX;
    ... // styles
`
```

File: src/containers/App/index.js (hunk 9 above)
Comment (rmolinamir) [reply src/containers/App/index.js:23]: @jorgebaralt Which components are those? We should fix that instead, this seems like a bandaid fix.

**Review by rmolinamir — CHANGES_REQUESTED** [review]
A couple of additional questions and small requests, almost there bro 👍

**Review by rmolinamir — APPROVED** [review]
Approved and merged 👍

**Pull request merged**